drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -119,8 +119,33 @@ void TM_EvtReceivedCb(TL_EvtPacket_t *hcievt)
 					     K_FOREVER);
 			break;
 		}
+/*
+ * https://github.com/zephyrproject-rtos/zephyr/issues/19509
+ * For some reason BLE stack (v 1.2.0) is not reporting the peer address
```

Comment:
The bug as it appears to me is that: "Unresolved Resolvable Private Addresses (RPA) is reported in the peer_rpa field, and not in the peer address, as it should, when this happens the peer address is set to all FFs"
Might be a better explanation for what you are doing here.